exporters/otlp/src/otlp_file_client.cc

```diff
@@ -0,0 +1,1587 @@
+        return std::shared_ptr<std::ofstream>();
+      }
+      of->close();
+    }
+
+    of->open(file_path, std::ios::binary | std::ios::out | std::ios::app);
+    if (!of->is_open())
+    {
+      OTEL_INTERNAL_LOG_ERROR("[OTLP FILE Client] Open " << static_cast<const char *>(file_path)
+                                                         << " failed: " << options_.file_pattern);
+      return std::shared_ptr<std::ofstream>();
+    }
+
+    of->seekp(0, std::ios_base::end);
+    file_->written_size = static_cast<size_t>(of->tellp());
+
+    file_->current_file    = of;
+    file_->last_checkpoint = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+    file_->file_path.assign(file_path, file_path_size);
+
+    // Create hardlink for alias
+#if !defined(FS_DISABLE_LINK)
+    if (!options_.alias_pattern.empty())
+    {
+      char alias_file_path[FileSystemUtil::kMaxPathSize + 1];
+      std::size_t file_path_len = FormatPath(alias_file_path, sizeof(alias_file_path) - 1,
+                                             options_.alias_pattern, file_->rotate_index);
+      if (file_path_len <= 0)
+      {
+        OTEL_INTERNAL_LOG_ERROR("[OTLP FILE Client] Generate alias file path from "
+                                << options_.alias_pattern << " failed");
+        return file_->current_file;
+      }
+
+      if (file_path_len < sizeof(alias_file_path))
+      {
+        alias_file_path[file_path_len] = 0;
+      }
+
+      if (0 == strcasecmp(file_path, alias_file_path))
+      {
+        return file_->current_file;
+      }
+
+      int res =
+          FileSystemUtil::Link(file_->file_path.c_str(), alias_file_path,
+                               static_cast<int32_t>(FileSystemUtil::LinkOption::kForceRewrite));
+      if (res != 0)
+      {
+        OTEL_INTERNAL_LOG_ERROR("[OTLP FILE Client] Link " << file_->file_path << " to "
+                                                           << alias_file_path
+                                                           << " failed, errno: " << res);
+#  if !defined(__CYGWIN__) && defined(_WIN32)
+        OTEL_INTERNAL_LOG_ERROR(
+            "[OTLP FILE Client]     you can use FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER | "
+            "FORMAT_MESSAGE_FROM_SYSTEM"
+            << " | FORMAT_MESSAGE_IGNORE_INSERTS, nullptr, " << res
+            << ", MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), "
+            << "(LPTSTR) &lpMsgBuf, 0, nullptr) to get the error message, see "
+            << "https://docs.microsoft.com/en-us/windows/desktop/api/WinBase/"
+               "nf-winbase-formatmessage and "
+            << "https://docs.microsoft.com/en-us/windows/desktop/Debug/"
+               "retrieving-the-last-error-code for more "
+               "details");
+#  else
+        OTEL_INTERNAL_LOG_ERROR("[OTLP FILE Client]     you can use strerror("
+                                << res << ") to get the error message");
```

Comment:
Same for non windows:

Do we really want to print these instructions in the log ?

How about just printing the `res` return code, and have these instructions as C++ comments.